python/tvm/relay/op/strategy/arm_cpu.py

```diff
@@ -232,7 +232,7 @@ def conv2d_strategy_arm_cpu(attrs, inputs, out_type, target):
                             wrap_topi_schedule(topi.arm_cpu.schedule_conv2d_NHWC_quantized_native),
                             name="conv2d_NHWC_quantized_native.arm_cpu",
                         )
-                    if is_aarch64 and has_asimd:
+                    if has_asimd:
```

Comment:
Right, at this point we can remove the `if has_asimd`condition completely, so I'll at least do that.  

From my understanding, the reason we are adding the implementations as "interleaved", "native", and then "interleaved" again is to make use of the fact that when two implementations have the same priority level, the one added to the strategy first is selected by default.  
We currently want the default to be:

- both i8mm & dotprod enabled: interleaved
- i8mm: interleaved
- dotprod: native
- neither (only aarch64): interleaved

We could replace the entire int8 strategy selection with  
```
if has_dot_prod and not has_matmul_i8:
    strategy.add_implementation(
        wrap_compute_conv2d(topi.arm_cpu.compute_conv2d_NHWC_quantized_native),
        wrap_topi_schedule(topi.arm_cpu.schedule_conv2d_NHWC_quantized_native),
        name="conv2d_NHWC_quantized_native.arm_cpu",
    )
strategy.add_implementation(
    wrap_compute_conv2d(
        topi.arm_cpu.compute_conv2d_NHWC_quantized_interleaved
    ),
    wrap_topi_schedule(
        topi.arm_cpu.schedule_conv2d_NHWC_quantized_interleaved
    ),
    name="conv2d_NHWC_quantized_interleaved.arm_cpu",
)
```
but I guess this wouldn't allow autoTVM to select between the 2 implementations when both i8mm and dotprod are enabled.  
Which way do you think is best?

Reply:
I'd favour making it explicit in the if else block which combination of architecture extensions results in which set of schedules added to the strategy, even though it would create more lines of code. So essentially recreating that bullet point table there, with both schedules added when both i8mm and dotprod are available, in the right priority order. What do you think?

Reply:
Sure, having a separate branch for each architecture extension combination should make it easier to tell at a glance which schedules it supports, at the cost of some repeated code.  
While I'm at it, I will make this change for the `conv2d_gemm_without_weight_transform_strategy_arm_cpu` strategy too.